zkevm-circuits/src/copy_circuit.rs

```diff
@@ -425,101 +429,117 @@ impl<F: Field> SubCircuitConfig<F> for CopyCircuitConfig<F> {
-                        CopyTable::assignments(copy_event, challenges)
-                            .iter()
-                            .enumerate()
-                    {
-                        let is_read = step_idx % 2 == 0;
-                        // Copy table assignments
-                        for (column, &(value, label)) in copy_table_columns.iter().zip_eq(table_row)
-                        {
-                            // Leave sr_addr_end and bytes_left unassigned when !is_read
-                            if !is_read && (label == "src_addr_end" || label == "bytes_left") {
-                            } else {
-                                region.assign_advice(
-                                    || format!("{} at row: {}", label, offset),
-                                    *column,
-                                    offset,
-                                    || value,
-                                )?;
-                            }
-                        }
-
-                        // q_step
-                        if is_read {
-                            self.q_step.enable(&mut region, offset)?;
-                        }
-                        // q_enable
-                        region.assign_fixed(
-                            || "q_enable",
-                            self.q_enable,
-                            offset,
-                            || Value::known(F::one()),
-                        )?;
-
-                        // is_last, value, is_pad, is_code
-                        for (column, &(value, label)) in
-                            [self.is_last, self.value, self.is_pad, self.is_code]
-                                .iter()
-                                .zip_eq(circuit_row)
-                        {
-                            region.assign_advice(
-                                || format!("{} at row: {}", label, offset),
-                                *column,
-                                offset,
-                                || value,
-                            )?;
-                        }
-
-                        //tag
-                        tag_chip.assign(&mut region, offset, tag)?;
-
-                        // lt chip
-                        if is_read {
-                            lt_chip.assign(
-                                &mut region,
-                                offset,
-                                F::from(
-                                    copy_event.src_addr + u64::try_from(step_idx).unwrap() / 2u64,
-                                ),
-                                F::from(copy_event.src_addr_end),
-                            )?;
-                        }
-
-                        offset += 1;
-                    }
+                for copy_event in copy_events.iter() {
+                    self.assign_copy_event(
+                        &mut region,
+                        offset,
```

Comment:
Consider passing `&mut offset` so that offset is incremented via the `assign_copy_event` without requiring the extra `offset += 2 * copy_event.bytes.len();`

Reply:
I like it - feels cleaner. I also changed the padding methods to take `&mut offset` as well.